### ISSUE TITLE
Incompatibility of the partitions dependency with future versions of Rust

### ISSUE THREAD

**Issue by gilad-rosenblatt** [rust-ml/linfa#299]
Would you be able to replac/update the `partitions v0.2.4` dependency for `linfa-clustering` so that I can continue using `linfa` with future versions of Rust? The existing version has some lint issues that are incompatible w/ future versions of Rust (see below).

# Description

While building my project I get a future incompatibility warning coming from `partitions v0.2.4` (a dependency introduced by `linfa-clustering`). 
```
warning: the following packages contain code that will be rejected by a future version of Rust: partitions v0.2.4
```

# Details

On running `cargo report future-incompatibilities --id 4` it seems the that `partitions v0.2.4` is expected to have some linting issues.
```The package `partitions v0.2.4` currently triggers the following future incompatibility lints:
> warning: trailing semicolon in macro used in expression position
>    --> /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/lib.rs:69:51
>     |
> 69  |         bit_vec::BitVec::from_elem($len, $element);
>     |                                                   ^
>     |
>    ::: /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/partition_vec.rs:528:24
>     |
> 528 |         let mut done = bit_vec![false; self.len()];
>     |                        --------------------------- in this macro invocation
>     |
>     = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>     = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>     = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>     = note: this warning originates in the macro `bit_vec` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>     --> /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/lib.rs:69:51
>      |
> 69   |         bit_vec::BitVec::from_elem($len, $element);
>      |                                                   ^
>      |
>     ::: /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/partition_vec.rs:1297:19
>      |
> 1297 |             done: bit_vec![false; len],
>      |                   -------------------- in this macro invocation
>      |
>      = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>      = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>      = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>      = note: this warning originates in the macro `bit_vec` (in Nightly builds, run with -Z macro-backtrace for more info)
> 
> warning: trailing semicolon in macro used in expression position
>     --> /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/lib.rs:69:51
>      |
> 69   |         bit_vec::BitVec::from_elem($len, $element);
>      |                                                   ^
>      |
>     ::: /home/maticd/.cargo/registry/src/github.com-1ecc6299db9ec823/partitions-0.2.4/src/partition_vec.rs:1345:19
>      |
> 1345 |             done: bit_vec![false; len],
>      |                   -------------------- in this macro invocation
>      |
>      = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
>      = note: for more information, see issue #79813 <https://github.com/rust-lang/rust/issues/79813>
>      = note: `#[allow(semicolon_in_expressions_from_macros)]` on by default
>      = note: this warning originates in the macro `bit_vec` (in Nightly builds, run with -Z macro-backtrace for more info)
```

**Comment by gilad-rosenblatt** [comment]
Just noticed this is a the same as #295.

**Comment by YuhanLiin** [comment]
Resolved by #298

**Comment by ThatGeoGuy** [comment]
This was purportedly fixed in #298 (per above comment) at the end of March. Is there a new release on the way that can finally solve this transient warning? 

What is the expected timeline for the next release? This project has moved on considerably as of the last release in December 2022 and we're 8 months since.
